### ISSUE TITLE
Reverse Proxy Server Replay fails (expected absolute, got relative URL)

### ISSUE THREAD

**Issue by mhils** [mitmproxy/mitmproxy#303]
https://groups.google.com/forum/#!topic/mitmproxy/RMEbEP7j_kc

Gist:
mitmproxy only accepts absolute-urls in regular proxy mode (GET http://example.com/). In reverse proxy mode, this changes to relative form (GET /), as a normal HTTP server would receive. This logic sits in front of everything else. The server replay mode is not aware that the recorded traffic has been in relative form and rightfully complains about the invalid request mode.
TL;DR: A very simple workaround is running mitmproxy -S revproxy.out -R http://127.0.0.1 (arbitrary reverse proxy URL). This sets mitmproxy in relative URL mode and everything works fine.

@cortesi: Do we want to change the behaviour or the documentation?

**Comment by hexsel** [comment]
Hm I tried this, but this seems to put the server back in proxy mode, so calls are not replayed from the file, but redirected back to the -R address.

mitmproxy 0.13

E.g. this line redirects all calls to http://localhost/<request>

  ./mitmdump -p $PROXY_PORT --server-replay REPLAY.dump -R http://localhost

**Comment by mhils** [comment]
@hexsel, what happens if you use the original address instead of localhost as the destination?

**Comment by hexsel** [comment]
it works, but only if the server is available (i.e. it will actually go back to proxying these requests instead of failing)

I've been trying to use this to avoid servers that I need to VPN in to access, but if the server is not available (i.e. VPN off), it won't even return the [replay] requests.

**Comment by hexsel** [comment]
however, it does work to some degree

**Comment by mhils** [comment]
Thanks. The issue of server availability is fixed on master, so it should work with the next release.

**Comment by hexsel** [comment]
awesome, thanks!

**Comment by mhils** [comment]
Nobody has complained anymore and I think the current behaviour is reasonable enough - closing for now.
